### PR TITLE
For #26089 - Record metrics for save credit card prompt shown

### LINKED ISSUE #26089: [Telemetry] Record metrics for save credit card prompt shown fact
Fenix counterpart of [12524](https://github.com/mozilla-mobile/android-components/pull/12524).

Record metrics for the facts emmited when the save credit card prompt is shown.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21054)

### PR DESCRIPTION
**Pull request by Alexandru2909**
Record metrics for the facts emmited when the save credit card prompt is shown.

Waiting for [12524](https://github.com/mozilla-mobile/android-components/pull/12524) to be merged.

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### GitHub Automation
Fixes #26089

### CODE REVIEW

**Comment by Alexandru2909** [comment]
<h1 dir="auto">Request for data collection review form</h1>
<p dir="auto">All questions are mandatory. You must receive review from a
 data steward peer on your responses to these questions before shipping 
new data collection.</p>
<ol dir="auto">
<li>What questions will you answer with this data?</li>
</ol>
<ul dir="auto">
<li>
    How many times is the credit card save prompt displayed?
</li>
<li>
    Are our users discovering the feature to save from autofill and are they finding it valuable? 
</li>
</ul>
<ol dir="auto" start="2">
<li>Why does Mozilla need to answer these questions? Are there benefits 
for users? Do we need this information to address product or business 
requirements?</li>
</ol>
<ul dir="auto">
<li>We need to know how useful credit card autofill is for the users and how 
we can best serve them when it comes to their shopping experiences. This
 data will help us establish the click-through-rates of saving and updating credit cards  as a way to measure feature success.</li>
</ul>
<ol dir="auto" start="3">
<li>What alternative methods did you consider to answer these questions? Why were they not sufficient?</li>
</ol>
<ul dir="auto">
<li>No other alternatives. </li>
</ul>
<ol dir="auto" start="4">
<li>Can current instrumentation answer these questions?</li>
</ol>
<ul dir="auto">
<li>No. </li>
</ul>
<ol dir="auto" start="5">
<li>List all proposed measurements and indicate the category of data 
collection for each measurement, using the Firefox data collection 
categories found on the Mozilla wiki. <br>
Note that the data steward reviewing your request will characterize your
 data collection based on the highest (and most sensitive) category.</li>
</ol>

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
Save prompt shown: The credit card autofill save prompt was displayed to the user. | Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/26089

<ol dir="auto" start="6">
<li>Please provide a link to the documentation for this data collection 
which describes the ultimate data set in a public, complete, and 
accurate way.</li>
</ol>
<ul dir="auto">
<li><a href="https://dictionary.telemetry.mozilla.org/apps/fenix" rel="nofollow">https://dictionary.telemetry.mozilla.org/apps/fenix </a></li>
</ul>
<ol dir="auto" start="7">
<li>How long will this data be collected? Choose one of the following:</li>
</ol>
<ul dir="auto">
<li>Up until version 105, with the option to renew at that point. This is also the version when the other credit card telemetry expires.</li>
</ul>
<ol dir="auto" start="8">
<li>What populations will you measure?</li>
</ol>
<ul dir="auto">
<li>All channels, all locales, all countries. </li>
</ul>
<ol dir="auto" start="9">
<li>If this data collection is default on, what is the opt-out mechanism for users?</li>
</ol>
<ul dir="auto">
<li>Default Glean SDK opt-out mechanism. </li>
</ul>
<ol dir="auto" start="10">
<li>Please provide a general description of how you will analyze this data.</li>
</ol>
<ul dir="auto">
<li>Glean. </li>
</ul>
<ol dir="auto" start="11">
<li>Where do you intend to share the results of your analysis?</li>
</ol>
<ul dir="auto">
<li> Only on Glean and with mobile teams. </li>
</ul>
<ol dir="auto" start="12">
<li>Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?</li>
</ol>
<ul dir="auto">
<li>No third-party tools.</li>
</ul>

**Comment by JohanLorenzo** [comment]
Rebasing PR because https://github.com/mozilla-mobile/fenix/pull/26073 reached `main`. It renamed the expected check from `pr-complete` to `complete-pr`. This impacts mergify and it may prevent you from merging this PR. That's why the rebase is needed. Let me know if you have any questions.

File: app/metrics.yaml [hunk 1]
```diff
@@ -7464,6 +7464,22 @@ credit_cards:
...
+      The autofill save prompt is shown.
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/26089
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/26095
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - android-probes@mozilla.com
+    expires: 105
```
Comment (gabrielluong) [inline app/metrics.yaml:7479]: 105 would mean this expires next release.

Comment (Alexandru2909) [reply app/metrics.yaml]: Changed that to 118, based on the telemetry expiry date renewal in https://github.com/mozilla-mobile/fenix/issues/26121.

File: app/src/test/java/org/mozilla/fenix/components/metrics/MetricControllerTest.kt [hunk 2]
```diff
@@ -583,7 +583,7 @@ class MetricControllerTest {
             CreditCardAutofillDialogFacts.Items.AUTOFILL_CREDIT_CARD_PROMPT_EXPANDED to CreditCards.autofillPromptExpanded,
             CreditCardAutofillDialogFacts.Items.AUTOFILL_CREDIT_CARD_PROMPT_DISMISSED to CreditCards.autofillPromptDismissed,
             CreditCardAutofillDialogFacts.Items.AUTOFILL_CREDIT_CARD_CREATED to CreditCards.savePromptCreate,
-            CreditCardAutofillDialogFacts.Items.AUTOFILL_CREDIT_CARD_UPDATED to CreditCards.savePromptUpdate,
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/components/metrics/MetricControllerTest.kt:586]: Did we want to remove this item event from being tested?

Comment (Alexandru2909) [reply app/src/test/java/org/mozilla/fenix/components/metrics/MetricControllerTest.kt:586]: It was not intentional. Fixed it now.

File: app/metrics.yaml [hunk 3]
```diff
@@ -7484,6 +7484,22 @@ credit_cards:
     metadata:
       tags:
         - Autofill
+  save_prompt_shown:
+    type: event
+    description: |
+      The autofill save prompt is shown.
```
Comment (gabrielluong) [inline app/metrics.yaml:7490]:
```suggestion
      The credit card autofill save prompt is shown.
```

**Review by gabrielluong — APPROVED** [review]

**Comment by rocketsroger** [comment]
> # Request for data collection review form
> 
> All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.
> 
>     1. What questions will you answer with this data?
> 
> 
>     * How many times is the credit card save prompt displayed?
> 
>     * Are our users discovering the feature to save from autofill and are they finding it valuable?
> 
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits 
>        for users? Do we need this information to address product or business 
>        requirements?
> 
> 
>     * We need to know how useful credit card autofill is for the users and how 
>       we can best serve them when it comes to their shopping experiences. This
>        data will help us establish the click-through-rates of saving and updating credit cards  as a way to measure feature success.
> 
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
>     * No other alternatives.
> 
> 
>     4. Can current instrumentation answer these questions?
> 
> 
>     * No.
> 
> 
>     5. List all proposed measurements and indicate the category of data 
>        collection for each measurement, using the Firefox data collection 
>        categories found on the Mozilla wiki.    
>        Note that the data steward reviewing your request will characterize your
>         data collection based on the highest (and most sensitive) category.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Save prompt shown: The credit card autofill save prompt was displayed to the user. 	Category 2 - interaction data 	#26089
> 
>     6. Please provide a link to the documentation for this data collection 
>        which describes the ultimate data set in a public, complete, and 
>        accurate way.
> 
> 
>     * [https://dictionary.telemetry.mozilla.org/apps/fenix   ](https://dictionary.telemetry.mozilla.org/apps/fenix)
> 
> 
>     7. How long will this data be collected? Choose one of the following:
> 
> 
>     * Up until version 118, with the option to renew at that point.
> 
> 
>     8. What populations will you measure?
> 
> 
>     * All channels, all locales, all countries.
> 
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
>     * Default Glean SDK opt-out mechanism.
> 
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
>     * Glean.
> 
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
>     * Only on Glean and with mobile teams.
> 
> 
>     12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
> 
> 
>     * No third-party tools.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

N/A, collection set to end or be renewed by version 118

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

**Pull request merged**